tests/unittests/test_tabular_bagstack.py

```diff
@@ -0,0 +1,176 @@
+    # otherwise we run full benchmark with default AutoGluon settings.
+    # performance_value warnings are disabled when fast_benchmark = True.
+
+    #### If fast_benchmark = True, can control model training time here. Only used if fast_benchmark=True ####
+    if fast_benchmark:
+        subsample_size = 100
+        nn_options = {'num_epochs': 3} 
+        gbm_options = {'num_boost_round': 30}
+        hyperparameters = {'GBM': gbm_options, 'NN': nn_options}
+        time_limits = 60
+
+    fit_args = {
+        'num_bagging_folds': num_bagging_folds,
+        'stack_ensemble_levels': stack_ensemble_levels,
+        'hyperparameter_tune': hyperparameter_tune,
+        'verbosity': verbosity,
+    }
+    if hyperparameters is not None:
+        fit_args['hyperparameters'] = hyperparameters
+    if time_limits is not None:
+        fit_args['time_limits'] = time_limits
+    ###################################################################
+    run_tabular_benchmarks(fast_benchmark=fast_benchmark, subsample_size=subsample_size, perf_threshold=perf_threshold,
+                           seed_val=seed_val, fit_args=fit_args)
+
+
+def test_tabular_bagstack():
+    ############ Benchmark options you can set: ########################
+    stack_ensemble_levels = 2
+    num_bagging_folds = 3
+    perf_threshold = 1.1 # How much worse can performance on each dataset be vs previous performance without warning
+    seed_val = 0 # random seed
+    subsample_size = None
+    hyperparameter_tune = False
+    verbosity = 2 # how much output to print
+    hyperparameters = None
+    time_limits = None
+    fast_benchmark = True # False
+    # If True, run a faster benchmark (subsample training sets, less epochs, etc),
+    # otherwise we run full benchmark with default AutoGluon settings.
+    # performance_value warnings are disabled when fast_benchmark = True.
+
+    #### If fast_benchmark = True, can control model training time here. Only used if fast_benchmark=True ####
+    if fast_benchmark:
+        subsample_size = 100
+        nn_options = {'num_epochs': 3} 
+        gbm_options = {'num_boost_round': 30}
+        hyperparameters = {'GBM': gbm_options, 'NN': nn_options, 'custom': ['GBM']}
+        time_limits = 60
+
+    fit_args = {
+        'num_bagging_folds': num_bagging_folds,
+        'stack_ensemble_levels': stack_ensemble_levels,
+        'hyperparameter_tune': hyperparameter_tune,
+        'verbosity': verbosity,
+    }
+    if hyperparameters is not None:
+        fit_args['hyperparameters'] = hyperparameters
+    if time_limits is not None:
+        fit_args['time_limits'] = time_limits
+    ###################################################################
+    run_tabular_benchmarks(fast_benchmark=fast_benchmark, subsample_size=subsample_size, perf_threshold=perf_threshold,
+                           seed_val=seed_val, fit_args=fit_args)
+
+
+if __name__ == '__main__':
```

Comment:
Seems like a lot of duplicate code here. I think it might be nice to reduce the duplication by having default parameter values and then only having to specify the non-standard parameters to the test func (stack_ensemble_levels and num_bagging_folds). This would avoid subtle bugs if we have to refactor this later.

This could be done by having another function run_tabular_benchmarks_template(), which would call run_tabular_benchmarks with all the code above, but having you able to specify overrides.

Reply:
I actually think we eventually want to change the defaults in the different tests, so we have more comprehensive test coverage of various scenarios. This is why the parameters are defined in every test, they just happen to be the same right now.  I'm not sure our our set of default base models is fully finalized, but once it is, I think these parameters should aim cover different subsets of base models in different unit tests. Because I'm confident GBM and NN will be in the official final set of base models, this is why these are currently the only ones tried for now.  I've gone and changed the seed for the different tests in new commit.